dali/pipeline/operators/resize/resize.cc

```diff
@@ -20,20 +20,37 @@ namespace dali {
-      "This option is mutually exclusive with `resize_shorter`. "
-      "If the `resize_y` is left at 0, then the op will keep "
-      "the aspect ratio of the original image.", 0.f, true)
-  .AddOptionalArg("resize_y", "The length of the Y dimension of the resized image. "
-      "This option is mutually exclusive with `resize_shorter`. "
-      "If the `resize_x` is left at 0, then the op will keep "
-      "the aspect ratio of the original image.", 0.f, true)
-  .AddOptionalArg("resize_shorter", "The length of the shorter dimension of the resized image. "
-      "This option is mutually exclusive with `resize_longer`, `resize_x` and `resize_y`. "
-      "The op will keep the aspect ratio of the original image.", 0.f, true)
-  .AddOptionalArg("resize_longer", "The length of the longer dimension of the resized image. "
-      "This option is mutually exclusive with `resize_shorter`,`resize_x` and `resize_y`. "
-      "The op will keep the aspect ratio of the original image.", 0.f, true);
+  .AddOptionalArg("resize_x", R"code(The length of the X dimension of the resized image.
+This option is mutually exclusive with `resize_shorter`.
+If the `resize_y` is left at 0, then the op will keep
+the aspect ratio of the original image.)code", 0.f, true)
+  .AddOptionalArg("resize_y", R"code(The length of the Y dimension of the resized image.
+This option is mutually exclusive with `resize_shorter`.
+If the `resize_x` is left at 0, then the op will keep
+the aspect ratio of the original image.)code", 0.f, true)
+  .AddOptionalArg("resize_shorter", R"code(The length of the shorter dimension of the resized image.
+This option is mutually exclusive with `resize_longer`, `resize_x` and `resize_y`.
+The op will keep the aspect ratio of the original image.
+The longer dimension can be bounded by setting the `max_size` argument.
+See `max_size` argument doc for more info.)code", 0.f, true)
+  .AddOptionalArg("resize_longer", R"code(The length of the longer dimension of the resized image.
+This option is mutually exclusive with `resize_shorter`,`resize_x` and `resize_y`.
+The op will keep the aspect ratio of the original image.)code", 0.f, true)
+  .AddOptionalArg("max_size", R"code(Maximum size of the longer dimension when resizing with `resize_shorter`.
+When set with `resize_shorter`, the shortest dimension will be resized to `resize_shorter` iff
+the longest dimension is smaller or equal to `max_size`. If not, the shortest dimension is resized to
+satisfy the constraint "longest_dim == `max_size`.
+Can be also an array of size 2, where the two elements are maximum size per dimension (H, W).
+
+Example:
+
+Original image = "400x1200".
+
+Resized with:
+
+* `resize_shorter`="200"  (`max_size` not set) => "200x600"
+* `resize_shorter`="200", `max_size`="400      => "132x400"
+* `resize_shorter`="200", `max_size`=1000      => "200x600")code", std::vector<float>{0.f, 0.f}, true);
```

Comment:
The schema accepts ArgumentInputs, but currently you can't get ArgumentInput with non-scalar values.
We can either have 3 arguments (`max_size`, `max_width`, `max_height`) and enforce that neither max_width and max_height is specified when max_size is (and vice versa) or forbid per-sample limit (which makes more sense IMO).

Reply:
I agree, for my use-case and the ones I can think of, forbidding per-sample arg makes sense.

Reply:
Done